src/sagemaker/workflow/utilities.py

```diff
@@ -363,8 +363,8 @@ def override_pipeline_parameter_var(func):
     We should remove this decorator after the grace period.
     """
     warning_msg_template = (
-        "The input argument %s of function (%s) is a pipeline variable (%s), which is not allowed. "
-        "The default_value of this Parameter object will be used to override it. "
+        "The input argument %s of function (%s) is a pipeline variable (%s), which will not work. "
```

Comment:
The obvious question here for any user will be: "why will it not work?"

We should get this reviewed by a doc writer to get the least confusing wording